core/api.py

```diff
@@ -119,3 +137,76 @@ def by_date(self, date:str = 'today', period: str = '1d' ) -> dict:
         url += f"/activities/heart/date/{date}/{period}.json"
         return api_request(url, self._tokens)
 
+    def by_interval_intraday(
+        self,
+        start_date: str = 'today',
+        end_date: str = 'today',
+        detail_level: str = '1min',
+        start_time: str = None,
+        end_time: str = None,) -> dict:
+        """Heart rate intraday by interval.
+
+        Time parameters are optional.
+
+        API endpoint returns inconsistent results depending on the parameters.
+        The following table documents the results of different parameter combinations:
+
+        start_date   | end_date     | detail_level | start_time | end_time | result
+        ---------------------------------------------------------------------
+        'today'      | 'today'      | '1min'       | None       | None     | daily + intraday
+        'today'      | 'today'      | '1sec'       | None       | None     | invalid 
+        '2022-08-14' | '2022-08-14' | '1min        | None       | None     | daily + intraday
+        '2022-08-14' | '2022-08-14' | '1sec        | None       | None     | generic activity categories
+        '2022-08-11' | '2022-08-14' | '1min"       | None       | None     | daily only
+        '2022-08-11' | '2022-08-14' | '1sec"       | None       | None     | generic activity categories
+        'today'      | 'today'      | '1min'       | '07:00'    | '13:00'  | daily + timeboxed intraday
+        'today'      | 'today'      | '1sec'       | '07:00'    | '13:00'  | daily + timeboxed intraday at 1 sec intervals
+        '2022-08-14' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | daily + timeboxed intraday
+        '2022-08-14' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | daily + timeboxed intraday at 1 sec intervals
+        '2022-08-11' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | invalid
+        '2022-08-11' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | invalid
+
+        Reference:
+        https://dev.fitbit.com/build/reference/web-api/heartrate-timeseries/get-heartrate-timeseries-by-date/
+
+        Args:
+            start_date (str, optional): yyyy-MM-dd format date, or 'today'.. Defaults to 'today'.

```

Comment:
Are the double periods are here and on the next line a mistake?

Reply:
done